src/main/java/org/breedinginsight/brapps/importer/daos/BrAPITrialDAO.java

```diff
@@ -58,109 +71,175 @@ public BrAPITrialDAO(ProgramDAO programDAO, ImportDAO importDAO, BrAPIDAOUtil br
+            return updatedTrial;
+        } catch (Exception e) {
+            throw new InternalServerException("Unknown error has occurred: " + e.getMessage(), e);
+        }
+    }
 
-        TrialsApi api = brAPIEndpointProvider.get(programDAO.getCoreClient(program.getId()), TrialsApi.class);
-        return processExperimentsForDisplay(brAPIDAOUtil.search(
-                api::searchTrialsPost,
-                api::searchTrialsSearchResultsDbIdGet,
-                trialSearch), program.getKey(), programId, true);
+    public List<BrAPITrial> getTrials(UUID programId) throws ApiException {
+        return new ArrayList<>(programExperimentCache.get(programId).values());
     }
 
-    //Removes program key from trial name and adds dataset information
-    private List<BrAPITrial> processExperimentsForDisplay(List<BrAPITrial> trials, String programKey, UUID programId, boolean metadata) throws ApiException {
+    //Removes program key from trial name
+    private List<BrAPITrial> processExperimentsForDisplay(
+            List<BrAPITrial> trials,
+            String programKey) throws ApiException {
         List<BrAPITrial> displayExperiments = new ArrayList<>();
         for (BrAPITrial trial: trials) {
             trial.setTrialName(Utilities.removeProgramKey(trial.getTrialName(), programKey, ""));
-            List<String> datasets = new ArrayList<>();
-
-            if (metadata) {
-                //todo presumably BI-1193 replace dummy value with list of datasets once datasets implemented
-                datasets.add("Observation Dataset");
-                trial.putAdditionalInfoItem("datasets", datasets);
-            }
-
             displayExperiments.add(trial);
         }
         return displayExperiments;
     }
 
-    public Optional<BrAPITrial> getTrialById(UUID programId, UUID trialDbId) throws ApiException, DoesNotExistException {
-        Program program = programService.getById(programId).orElseThrow(() -> new DoesNotExistException("Program id does not exist"));
-        String refSoure = Utilities.generateReferenceSource(BRAPI_REFERENCE_SOURCE, ExternalReferenceSource.TRIALS);
-        List<BrAPITrial> trials = getTrialsByExRef(refSoure, trialDbId.toString(), program);
+    public Optional<BrAPITrial> getTrialById(UUID programId, UUID trialId) throws ApiException, DoesNotExistException {
+        Map<String, BrAPITrial> cache = programExperimentCache.get(programId);
+        BrAPITrial trial = null;
+        if (cache != null) {
+            trial = cache.get(trialId.toString());
+        }
 
-        return Utilities.getSingleOptional(trials);
+        return Optional.ofNullable(trial);
     }
 
     public Optional<BrAPITrial> getTrialByDbId(String trialDbId, Program program) throws ApiException {
         List<BrAPITrial> trials = getTrialsByDbIds(List.of(trialDbId), program);
         return Utilities.getSingleOptional(trials);
     }
```

Comment:
Does it make sense to either
1. remove this unused method, or
2. update it to utilize the cache?

Reply:
It already uses the cache via`getTrialsByDbIds`. As for the unused methods, it was decided in discussions with Tim to leave these in place for now.